File: app/src/main/java/news/reddit/com/redditnews/NewsViewModel.java
package news.reddit.com.redditnews;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;

import java.util.List;

import news.reddit.com.redditnews.model.NewsListDataSource;
import news.reddit.com.redditnews.network.ServiceGenerator;
import news.reddit.com.redditnews.response.NewsEntity;

public class NewsViewModel extends AndroidViewModel {

    private LiveData<List<NewsEntity>> observable;

    public NewsViewModel(Application application) {
        super(application);
        observable = new NewsListDataSource(new ServiceGenerator().getNewsApiSet(application.getApplicationContext())).getNews("a");
    }

    public LiveData<List<NewsEntity>> getNews() {
        return observable;
    }
}

File: app/src/main/java/news/reddit/com/redditnews/NewsViewModel.kt
package news.reddit.com.redditnews

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData

import news.reddit.com.redditnews.response.NewsEntity

class NewsViewModel(application: Application, private val repository: NewsListRepository) : AndroidViewModel(application) {

    private var newsData: LiveData<List<NewsEntity>>? = null

    fun getNews(): LiveData<List<NewsEntity>> {
        if (newsData == null){
            newsData = MutableLiveData<List<NewsEntity>>()
            loadNews()
        }
        return newsData as LiveData<List<NewsEntity>>
    }

    private fun loadNews() {
        newsData = repository.getNews("a")
    }

}
